resources/js/components/ui/calendar/ResourcesSelector.tsx

```diff
@@ -0,0 +1,130 @@
+import {cx, htmlAttributes} from "components/utils";
+import {For, JSX, Show, VoidComponent, createMemo, createRenderEffect, createSignal, splitProps} from "solid-js";
+
+export type ResourcesSelectionMode = "radio" | "checkbox";
+
+interface Props extends htmlAttributes.div {
+  resourceGroups?: readonly ResourceGroup[];
+  mode: ResourcesSelectionMode;
+  selection: readonly string[];
+  setSelection: (ids: string[]) => void;
+}
+
+export interface ResourceGroup {
+  readonly label: () => JSX.Element;
+  readonly resources: readonly Resource[];
+}
+
+export interface Resource {
+  readonly id: string;
+  readonly label: () => JSX.Element;
+}
+
+/**
+ * A vertical list of people and resources, with checkboxes or radio buttons, depending on mode.
+ */
+export const ResourcesSelector: VoidComponent<Props> = (allProps) => {
+  const [props, divProps] = splitProps(allProps, ["resourceGroups", "mode", "selection", "setSelection"]);
+  const selected = createMemo(() => new Set(props.selection));
+  let div: HTMLDivElement | undefined;
+  return (
+    <div ref={div} {...htmlAttributes.merge(divProps, {class: "flex flex-col gap-2"})}>
+      <For each={props.resourceGroups}>
+        {({label, resources}, i) => {
+          const inputId = () => (props.mode === "checkbox" ? `resourceGroup_${i()}` : undefined);
```

Comment:
hehe, no chat ma pewnie rację, że to może być raz zadeklarowane na górze jako jedna funkcja inputId(i)

Reply:
No nie wiem, nie wydaje mi się po pierwsze że to robi różnicę, a po drugie grup to są tak coś ze dwie :) więc w ogóle nie ma o czym gadać.